pyflakes/test/test_type_annotations.py

```diff
@@ -507,6 +507,51 @@ def f(x: Literal['some string']) -> None:
             return None
         """)
 
+    @skipIf(version_info < (3,), 'new in Python 3')
+    def test_forward_ref_valid(self):
+        self.flakes("""
+        from typing import Optional
+
+        def f(x: Optional['int']) -> None:
+            return None
+        """)
+
+    @skipIf(version_info < (3,), 'new in Python 3')
+    def test_forward_ref_missing(self):
+        self.flakes("""
+        from typing import Optional
+
+        def f(x: Optional['integer']) -> None:
+            return None
+        """, m.UndefinedName)
+
+    @skipIf(version_info < (3, 6), 'new in Python 3.6')
+    def test_annotated_type_typing_missing_forward_type(self):
+        self.flakes("""
+        from typing import Annotated
+
+        def f(x: Annotated['integer']) -> None:
+            return None
+        """, m.UndefinedName)
+
+    @skipIf(version_info < (3, 6), 'new in Python 3.6')
+    def test_annotated_type_typing_missing_forward_type_multiple_args(self):
+        self.flakes("""
+        from typing import Annotated
+
+        def f(x: Annotated['integer', 1]) -> None:
+            return None
+        """, m.UndefinedName)
+
+    @skipIf(version_info < (3, 6), 'new in Python 3.6')
+    def test_annotated_type_typing_with_string_args(self):
+        self.flakes("""
+        from typing import Annotated
+
+        def f(x: Annotated[int, '> 0']) -> None:
+            return None
+        """)
```

Comment:
the decorators on these three tests are incorrect -- there's nothing python3.6-specific about these particular tests

Reply:
Thanks, I had recalled wrongly that this was supported only for python 3.6 onwards. But it is supported for python >= 3.4. See https://github.com/python/typing/pull/637